docs/integrating-uptest-for-e2e-testing.md

```diff
@@ -0,0 +1,143 @@
+# Integrating Uptest for End to End Testing
+
+In this tutorial, we will integrate [uptest](https://github.com/upbound/uptest) to a Github repository to automate end to end
+testing managed resources. While we will use a _provider_ repository as an example, the process will be identical for a
+configuration repository.
+
+Starting with a provider repository with no end to end testing capability, we will end up having:
+- A make target to locally test examples end to end
+- A GitHub action triggered for PRs whenever a comment as `/test-examples=<examples-path>` is left
+
+## Setting up the Make targets
+
+1. Go to the [demo repository] which contains a GitHub provider generated using upjet and hit the `Use this template` button
```

Comment:
trying to use https://github.com/turkenh/demo-uptest-integration in this context, looks like this repo is not a template